dashboard/generate.py

```diff
@@ -137,6 +139,36 @@ def efu_get_version(self, ipaddr, port):
             self.dprint("connection reset (by peer?)")
             return "connection reset (by peer?)"
 
+
+    def check_fw_pipeline(self, ipaddr, port):
+        if self.test:
+            return 5
+        try:
+            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            s.connect((ipaddr, port))
+            s.shutdown(socket.SHUT_WR) # no data sent
+            data = b""
+            while True:
+                chunk = s.recv(4096)
+                if not chunk:
+                    break
+                data += chunk
+            lines = data.decode("utf-8", errors="ignore").strip().splitlines()
+            for line in lines:
+                try:
+                    obj = json.loads(line)
+                    # Look for any key ending with '.worker_state'
+                    for key, value in obj.items():
+                        if key.endswith(".worker_state"):
+                            return int(value)
+                except Exception:
+                    continue
```

Comment:
```suggestion
            for line in lines:
                obj = json.loads(line)
                # Look for any key ending with '.worker_state'
                for key, value in obj.items():
                    if key.endswith(".worker_state"):
                        return int(value)
```

Reply:
this `try/except` block isn't necessary. When the `line` is `{}`, json.loads works fine and the inner for loop is simply skipped (no items, empty list). The only potential issue is a ValueError when converting to int but I’d rather let that fail. The error message will clearly show that the server sent something unexpected.

```suggestion
            for line in lines:
                obj = json.loads(line)
                # Look for any key ending with '.worker_state'
                for key, value in obj.items():
                    if key.endswith(".worker_state"):
                        return int(value)
```

Reply:
Thank you for the input! 🙌 
I'll have it in mind going forward

I actually didn't want to have this to be reviewed since I'm yet to have it working (even with these changes). I'll have the PR as DRAFT until it's ready to be reviewed. Sorry not making it clear 😞